docs/developer/architecture/logs.rst

```diff
@@ -0,0 +1,215 @@
+Elasticsearch), MetalK8s should provide some documentation in order guide the
+customer to deploy and configure its own log collection agent.
+
+The following requirements focus on application logs. Audit logs are not part
+of the requirements.
+
+Requirements
+------------
+
+*Lightweight*
+
+A lightweight tool to store and expose logs is required in order to minimize
+the HW footprints (CPU, RAM, Disks):
+
+- limited history: Storing the logs for very large period (3 years) is not
+  something metalK8s needs to provide as a feature. This can be achieved using
+  external log centralization system.
+
+- stable ingestion: It is important to guarantee stable ingestion of the logs
+  and less important to guarantee stable performances when browsing/searching
+  the logs. However, peak loads related to complex logs queries should not
+  impact the application workloads and deploying log storage and search service
+  on infra nodes might help achieving this isolation.
+
+- stream indexing: It is not required to have automatic indexing of logs
+  content. Instead, the log centralization service should offer basic features
+  to group/filter logs per tag/metadata defining the log stream.
+
+*Accessible from a central UI/API*
+
+Platform Admin or Storage Admin can visualize logs from all containers in all
+namespaces as well as journal logs, including (kubelet, containerd, salt-minion
+, kernel, initrd, services, etc ...) in Grafana. One can correlate logs and
+metrics or alerts in one single Grafana Dashboard.
+Browsing logs can be achieved through a documented API in order to expose logs
+in MetalK8s UIs or other workloads UIs if needed.
+
+*Persistence, Retention*
+
+Logs should be stored on a persistent storage. Platform Administrator can
+configure a max retention period. Some automatic purging mechanism is triggered
+when logs are older than the retention period or when the persistent storage is
+about to reach its capacity limit. Purging jobs are logged.
+A typical and default retention period is of 2 weeks.
+A formula can be used by solution developers in order to properly size the
+persistent storage for log centralization (cf documentation requirement).
+
+*Horizontally scalable (capacity and performance)*
+
+The Platform Administrator can scale the service in order to ingest/query and
+store more logs. It can be because more workloads are running on the platform
+or because there is a need to keep bigger history of logs.
+
+*Highly Available*
+
+Log collection, ingestion, storage and query services can be replicated in
+order to ensure that we can lose at least one server in the cluster without
+impacting availability and reliability of the service.
+
+*Log Querying*
+
+Get all logs for a given period, node(s), pod regex, limited list of predefined
+labels and free keywords text.
+Typical Zenko use case: collecting all logs across several components, related
+to a S3 uniq request.
+Typical predefined labels are severity and namespace.
```

Comment:
I am probably late to the party ; but this could lead to requirements on the server producing the logs themselves (like "structured logs" or using the same labels/names across the stack). Not exactly a requirement on the system itself, more of a pre-requisite, IMO